### ISSUE TITLE
Can't build on Ubuntu 16.04

### ISSUE THREAD

**Issue by meownoid** [lamdu/lamdu#69]
```
   /home/egor/Work/GitHub/lamdu/.stack-work/downloaded/tY0zL32R4CX3/.stack-work/dist/x86_64-linux/Cabal-1.22.5.0/setup/setup --builddir=.stack-work/dist/x86_64-linux/Cabal-1.22.5.0 build lib:nodejs-exec --ghc-options " -ddump-hi -ddump-to-file"

    Process exited with code: ExitFailure 1
    Logs have been written to: /home/egor/Work/GitHub/lamdu/.stack-work/logs/nodejs-exec-0.1.0.0.log
    Preprocessing library nodejs-exec-0.1.0.0...
    In-place registering nodejs-exec-0.1.0.0...
    build_node.sh: 4: build_node.sh: Bad substitution
    build_node.sh: 1: build_node.sh: node: not found
    build_node.sh: 8: build_node.sh: version: not found
    build_node.sh: 8: build_node.sh: version: not found
    Using existing system node 
    cp: missing destination file operand after 'node/node'
    Try 'cp --help' for more information.
```

Node.js installed from ubuntu repos:

```
$ nodejs -v
v4.2.6
```

**Comment by yairchu** [comment]
Can you try downloading and building https://github.com/lamdu/nodejs-exec?
I suppose it will fail in the same way, but then we could check things which might fix it, like maybe running build_node.sh within with "bash" rather than "sh" fixes anything? (I'm shooting in the dark here as I don't have an Ubuntu around..)

**Comment by meownoid** [comment]
Attempting to build nodejs-exec gives me exactly the same error:
```
In-place registering nodejs-exec-0.1.0.0...
build_node.sh: 4: build_node.sh: Bad substitution
build_node.sh: 1: build_node.sh: node: not found
build_node.sh: 8: build_node.sh: version: not found
build_node.sh: 8: build_node.sh: version: not found
Using existing system node 
cp: missing destination file operand after 'node/node'
```
But `bash build_node.sh` seems to work. My `sh` is pointing to `zsh`.

**Comment by yairchu** [comment]
In that case, does changing it to use `"bash"` in `rawSystemExit verbosity "sh" ["build_node.sh"]` in nodejs-exec's `Setup.hs` make `stack build` work for it? In that case that should probably be the fix

**Comment by xavier83** [comment]
Unable to build on arch too. got the following build error during `stack install`

```
             
--  While building package Lamdu-0.1 using:
      /home/user/.stack/setup-exe-cache/x86_64-linux/setup-Simple-Cabal-1.22.5.0-ghc-7.10.3 --builddir=.stack-work/dist/x86_64-linux/Cabal-1.22.5.0 build lib:Lamdu exe:lamdu --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure (-6)
    Logs have been written to: /media/Data/Devel/lamdu/.stack-work/logs/Lamdu-0.1.log

    Preprocessing library Lamdu-0.1...
    In-place registering Lamdu-0.1...
    Preprocessing executable 'lamdu' for Lamdu-0.1...
    [  2 of 120] Compiling Lamdu.Compiler.Flatten ( Lamdu/Compiler/Flatten.hs, .stack-work/dist/x86_64-linux/Cabal-1.22.5.0/build/lamdu/lamdu-tmp/Lamdu/Compiler/Flatten.o )
    src/tcmalloc.cc:283] Attempt to free invalid pointer 0x14a8440
Completed 5 action(s).
```

**Comment by yairchu** [comment]
@xavier83 there's a separate issue for arch - https://github.com/lamdu/lamdu/issues/51
Sadly that one seems like a touch nut to track..

**Comment by yairchu** [comment]
This one was probably fixed in 25f8b2cf8ac2242273672efa8043b7e2e6c9fc82. @meownoid could you verify things work now?
